src/github.com/getlantern/connpool/connpool.go

```diff
@@ -20,17 +20,20 @@ var (
 
 type DialFunc func() (net.Conn, error)
 
-// Pool is a pool of connections.  Connections are pooled eagerly up to MinSize
-// and expire after ClaimTimeout.  Pool attempts to always have MinSize
-// unexpired connections ready to go so that callers don't have to wait on a
-// connection being established when they're ready to use it.
+// Pool is a pool of connections. Connections are pooled lazily up to Size and
+// expire after ClaimTimeout. Lazily here means that the Pool won't start to
+// fill until the first request for a connection. As long as the Pool is
+// actively being used, it will attempt to always have Size number of
+// connections ready to use.
 type Pool struct {
-	// MinSize: the pool will always attempt to maintain at least these many
+	// Size: while active, the pool will attempt to maintain at these many
 	// connections.
-	MinSize int
+	Size int
 
 	// ClaimTimeout: connections will be removed from pool if unclaimed for
-	// longer than ClaimTimeout.  The default ClaimTimeout is 10 minutes.
+	// longer than ClaimTimeout. The default ClaimTimeout is 10 minutes. Once
+	// connections have been removed from the Pool, they won't be replaced until
+	// another connection is requested, at which point the pool will fill again.
 	ClaimTimeout time.Duration
```

Comment:
Although changing `ClaimTimeout` does affect runtime behaviour, it is not supposed to be changed, and it's not atomic. Same for `RedialDelayIncrement` and `MaxRedialDelay`.